### PR TITLE
Test branch

### PR DESCRIPTION
**Pull request by asfandalidal**

### CODE REVIEW

File: app/src/main/java/com/example/myweather/main/WeatherRepositoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.example.myweather.main
+
+import com.example.myweather.model.WeatherResponse
+import com.example.myweather.network.ApiService
+
+class WeatherRepositoryImpl(private val apiService: ApiService) : WeatherRepository {
```
Comment (dastageerb) [inline app/src/main/java/com/example/myweather/main/WeatherRepositoryImpl.kt:6]: f8a7cf4e052a8672bb9cbc617057dff ? what is this ? authToken ?

File: app/src/main/java/com/example/myweather/main/WeatherRepository.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.example.myweather.main
+
+import com.example.myweather.model.WeatherResponse
+import kotlinx.coroutines.flow.Flow
+
+
+interface WeatherRepository {
+    suspend fun getCityData(cityName: String): WeatherResponse
```
Comment (dastageerb) [inline app/src/main/java/com/example/myweather/main/WeatherRepository.kt:8]: how would you handle the error state if response is null ?

File: app/src/main/java/com/example/myweather/main/WeatherViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+class WeatherViewModel @Inject constructor(
+    private val weatherRepository: WeatherRepository
+) : ViewModel() {
+    private val _weatherResponse = MutableStateFlow<WeatherResponse?>(null)
+    val weatherResponse: StateFlow<WeatherResponse?> = _weatherResponse
+
+    private val _errorMessage = MutableStateFlow<String?>(null)
+    val errorMessage: StateFlow<String?> = _errorMessage
+
+    fun setSearchQuery(query: String, isInitialFetch: Boolean = false) {
```
Comment (dastageerb) [inline app/src/main/java/com/example/myweather/main/WeatherViewModel.kt:24]: error handling needs improvement

Comment (dastageerb) [inline app/src/main/java/com/example/myweather/main/WeatherViewModel.kt:24]: no loading state from viewModel ?  use an enum or sealed class

**Comment by asfandalidal** [comment]
Yes API token

**Comment by asfandalidal** [comment]
ok i will handle it

**Comment by asfandalidal** [comment]
ok sir🫠

**Pull request merged**